### DISCUSSION TITLE
How to help set properties of workloads/traits in Appfile?

### DISCUSSION THREAD

**Discussion by zzxwill** [kubevela/kubevela#880] [Ideas] [answered]
Per [KubeVela learning Appfile Step by step](https://kubevela.io/#/en/developers/learn-appfile), developers can quickly write an Appfile with a default workload to deploy an application. But for other workloads like task/worker, and traits like autoscale/route/rollout, we have to look into official capability reference docs on how to set properties like https://kubevela.io/#/en/developers/references/traits/autoscale?id=properties.

![image](https://user-images.githubusercontent.com/2805315/104336070-f6a81800-552e-11eb-8608-c1aa16db67da.png)

We hope to have a cli command to quickly look up the reference docs of a workload/trait.

Here are some options.

- show reference information in cli

First we have to come up with a command to do the work. **Which command would you recommend?**

Second, display the reference information in a neat way.

```shell
$ vela xxx -autoscale
Name | Description | Type | Required | Default 
------------ | ------------- | ------------- | ------------- | ------------- 
 min | Minimal replicas of the workload | int | true |  
 max | Maximal replicas of the workload | int | true |  
 cpuPercent | Specify the value for CPU utilization, like 80, which means 80% | int | false |  
 cron | Cron type auto-scaling. Just for `appfile`, not available for Cli usage | [cron](#cron) | false |  
```

- redirect properties reference in a newly opened browser tab
```shell
$ vela workloads --usage
```
Directly open a new tab on a browser and display the page like https://kubevela.io/#/en/developers/references/traits/autoscale like the action of command `copilot docs`.

Which option do you prefer? Are there any other solution you would recommend? Thanks.

\\cc @wonderflow

**Comment by resouer** [comment, marked as answer]
@zzxwill 请先解决生成出的文档保存在哪里的问题，直接放在本地远端用户是/上层产品看不到的。另外 hack/doc-gen 应该是“能力安装”这个流程的一个自动化环节，而不是手动执行。
